pyuvdata/uvbeam/uvbeam.py

```diff
@@ -972,36 +1174,64 @@ def efield_to_power(
-                                * np.conj(efield_data[0, :, pair[1]])
-                            )
-                            * beam_object.basis_vector_array[0, comp_i] ** 2
-                            + (
-                                efield_data[1, :, pair[0]]
-                                * np.conj(efield_data[1, :, pair[1]])
-                            )
-                            * beam_object.basis_vector_array[1, comp_i] ** 2
-                            + (
-                                efield_data[0, :, pair[0]]
-                                * np.conj(efield_data[1, :, pair[1]])
-                                + efield_data[1, :, pair[0]]
-                                * np.conj(efield_data[0, :, pair[1]])
+                        if beam_object.future_array_shapes:
+                            power_data[0, pol_i] += (
+                                (
+                                    efield_data[0, pair[0]]
+                                    * np.conj(efield_data[0, pair[1]])
+                                )
+                                * beam_object.basis_vector_array[0, comp_i] ** 2
+                                + (
+                                    efield_data[1, pair[0]]
+                                    * np.conj(efield_data[1, pair[1]])
+                                )
+                                * beam_object.basis_vector_array[1, comp_i] ** 2
+                                + (
+                                    efield_data[0, pair[0]]
+                                    * np.conj(efield_data[1, pair[1]])
+                                    + efield_data[1, pair[0]]
+                                    * np.conj(efield_data[0, pair[1]])
+                                )
+                                * (
+                                    beam_object.basis_vector_array[0, comp_i]
+                                    * beam_object.basis_vector_array[1, comp_i]
+                                )
                             )
-                            * (
-                                beam_object.basis_vector_array[0, comp_i]
-                                * beam_object.basis_vector_array[1, comp_i]
+                        else:
+                            power_data[0, :, pol_i] += (
+                                (
+                                    efield_data[0, :, pair[0]]
+                                    * np.conj(efield_data[0, :, pair[1]])
+                                )
+                                * beam_object.basis_vector_array[0, comp_i] ** 2
+                                + (
+                                    efield_data[1, :, pair[0]]
+                                    * np.conj(efield_data[1, :, pair[1]])
+                                )
+                                * beam_object.basis_vector_array[1, comp_i] ** 2
+                                + (
+                                    efield_data[0, :, pair[0]]
```

Comment:
again, you can do the `...` here -- it works even if you do something like

```python
a = np.zeros((10, 11))
a[0, ..., 1]
```

i.e. when the dots represent a dimension that's not there, it will be fine.

Reply:
same problem here